File: LeetCode/167两数之和Ⅱ输入有序数组.go
func twoSum(numbers []int, target int) []int {
	hash := make(map[int]int, len(numbers))
	for k, v := range numbers {
		hash[v] = k
	}
	for k, v := range numbers {
		if val, ok := hash[target-v]; ok {
			return []int{k + 1, val + 1}
		}
	}
	return []int{0, 0}
}

File: LeetCode/167两数之和Ⅱ输入有序数组.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < numbers.length; i++) {
            map.put(numbers[i], i);
        }
        for (int i = 0; i < numbers.length; i++) {
            if (map.containsKey(target - numbers[i])) {
                return new int[]{i + 1, map.get(target - numbers[i]) + 1};
            }
        }
        return new int[]{0, 0};
    }
}